src/nvim/tempfile.c

```diff
@@ -55,28 +55,48 @@ static void vim_maketempdir(void)
   }
 }
 
-/// Delete the temp directory and all files it contains.
-void vim_deltempdir(void)
+/// Delete "name" and everything in it, recursively.
+/// @param name The path which should be deleted.
+/// @return 0 for success, -1 if some file was not deleted.
+int delete_recursive(char_u *name)
```

Comment:
@jbradaric Sure, if you wouldn't mind. I started to do it but got distracted by a segfault which needed fixing :)